test/index.test.ts

```diff
@@ -106,47 +106,48 @@ describe("mkdist", () => {
     expect(writtenFiles.sort()).toEqual(
       [
         "dist/README.md",
-        "dist/bar.d.ts",
+        "dist/bar.d.mts",
         "dist/bar.mjs",
         "dist/demo.css",
-        "dist/dir-export.d.ts",
+        "dist/dir-export.d.mts",
         "dist/dir-export.mjs",
         "dist/foo.mjs",
         "dist/foo.d.ts",
+        "dist/foo.d.mts",
         "dist/index.mjs",
-        "dist/index.d.ts",
+        "dist/index.d.mts",
         "dist/star/index.mjs",
-        "dist/star/index.d.ts",
+        "dist/star/index.d.mts",
         "dist/star/other.mjs",
-        "dist/star/other.d.ts",
+        "dist/star/other.d.mts",
         "dist/types.d.ts",
         "dist/components/index.mjs",
-        "dist/components/index.d.ts",
+        "dist/components/index.d.mts",
         "dist/components/blank.vue",
-        "dist/components/blank.vue.d.ts",
+        "dist/components/blank.vue.d.mts",
         "dist/components/js.vue",
-        "dist/components/js.vue.d.ts",
+        "dist/components/js.vue.d.mts",
```

Comment:
The declaration file of vue is special, perhaps it should be `.vue.d.ts`?

Reply:
Good point, but it looks like it should actually be `.d.vue.ts` https://www.typescriptlang.org/tsconfig/#allowArbitraryExtensions

Reply:
Declaration for vue has a longer history than this flag, and the most of vue community is currently using `.vue.d.ts`. 

I think it would be fine that just skip vue declaration. So that we can automatic switch to `.vue.d.ts` if the community decides to switch because we are also using `vue-tsc` to generate vue declaration

Reply:
Yes, I completely agree. I'll make sure those files stay untouched as part of this change and follow up with a separate issue.

Reply:
https://github.com/unjs/mkdist/issues/270